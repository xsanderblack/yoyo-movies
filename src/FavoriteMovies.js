import React, { Component } from 'react';
import store from 'store';
import MoviesList from './MoviesList';

// Component displays list of favorite movies from local storage.
export default class FavoriteMovies extends Component {
  state = {
    movies: []
  };

  // Method gets favorite movies from local storage and calls setFavoriteMovies method to populate state. 
  componentDidMount() {
    const movies = [];

    // We are finding favorite movies with help of favoriteMovie identifier,
    // wich we added in ToggleFav component.
    store.each((value) => {
      if (value.favoriteMovie) movies.push(value);
    })

    this.setFavoriteMovies(movies);
  }

  setFavoriteMovies = (movies) => {
    this.setState({ movies });
  };

  // Method will be called in ToggleFav component to remove movie from state after unfavorite action.
  onRemoveFromFavorits = (id) => {
    const { movies } = this.state;
    let indexToRemove;

    movies.forEach((movie, index) => {
      if (movie.id === id) indexToRemove = index;
    });
    movies.splice(indexToRemove, 1);

    this.setState({ movies });
  };

  render() {
    const { movies } = this.state;
    return (
      <MoviesList movies={movies} onRemoveFromFavorits={this.onRemoveFromFavorits} />
    );
  }
};
