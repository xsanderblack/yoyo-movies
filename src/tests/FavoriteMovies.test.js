import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import FavoriteMovies from '../FavoriteMovies';

it('FavoriteMovies renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FavoriteMovies />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render FavoriteMovies correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<FavoriteMovies />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
