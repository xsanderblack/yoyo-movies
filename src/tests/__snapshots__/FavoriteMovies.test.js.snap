// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FavoriteMovies correctly 1`] = `
<MoviesList
  movies={Array []}
  onRemoveFromFavorits={[Function]}
/>
`;
